keanu-project/src/test/java/io/improbable/keanu/e2e/regression/RadonHierarchicalRegression.java

```diff
@@ -141,41 +122,44 @@ private RegressionModel createSubModel(double[] allFloor,
+        Vertex muAlpha = bayesianNetwork.getVertexByLabel(new VertexLabel("MuIntercept"));
+        Vertex sigmaAlpha = bayesianNetwork.getVertexByLabel(new VertexLabel("SigmaIntercept"));
+        Vertex muBeta = bayesianNetwork.getVertexByLabel(new VertexLabel("MuGradient"));
+        Vertex sigmaBeta = bayesianNetwork.getVertexByLabel(new VertexLabel("SigmaGradient"));
+
+        NetworkSamples posteriorSamples = NUTS.builder()
+            .maxTreeHeight(5)
+            .saveStatistics(true)
+            .build()
+            .getPosteriorSamples(bayesianNetwork, Arrays.asList(muAlpha, muBeta, sigmaAlpha, sigmaBeta), 500).downSample(100);
+
+        double averagePosteriorMuA = posteriorSamples.getDoubleTensorSamples(muAlpha).getAverages().scalar();
+        double averagePosteriorSigmaA = posteriorSamples.getDoubleTensorSamples(sigmaAlpha).getAverages().scalar();
+        double averagePosteriorMuB = posteriorSamples.getDoubleTensorSamples(muBeta).getAverages().scalar();
+        double averagePosteriorSigmaB = posteriorSamples.getDoubleTensorSamples(sigmaBeta).getAverages().scalar();
+
+        assertThat(averagePosteriorMuB, allOf(greaterThan(-0.9), lessThan(-0.4 )));
+        assertThat(averagePosteriorMuA, allOf(greaterThan(1.2), lessThan(1.8 )));
+
+        assertThat(averagePosteriorSigmaB, allOf(greaterThan(0.), lessThan(0.5 )));
+        assertThat(averagePosteriorSigmaA, allOf(greaterThan(0.), lessThan(0.5 )));
+
+        System.out.println("done");
+    }
+
+    private void optimise(BayesianNetwork bayesianNetwork) {
         bayesianNetwork.probeForNonZeroProbability(100);
         GradientOptimizer optimizer = KeanuOptimizer.Gradient.builderFor(bayesianNetwork)
             .absoluteThreshold(0.25)
             .maxEvaluations(10000)
             .build();
         optimizer.maxAPosteriori();
-
-        assertValuesAreCorrect(bayesianNetwork, models);
-    }
-
-    private void assertValuesAreCorrect(BayesianNetwork bayesianNetwork, List<RegressionModel> models) {
-        DoubleVertex muIntercept = (DoubleVertex) bayesianNetwork.getVertexByLabel(new VertexLabel("MuIntercept"));
-        DoubleVertex muGradient = (DoubleVertex) bayesianNetwork.getVertexByLabel(new VertexLabel("MuGradient"));
-        DoubleVertex sigmaIntercept = (DoubleVertex) bayesianNetwork.getVertexByLabel(new VertexLabel("SigmaIntercept"));
-        DoubleVertex sigmaGradient = (DoubleVertex) bayesianNetwork.getVertexByLabel(new VertexLabel("SigmaGradient"));
-
-        Assert.assertTrue(-0.4 > muGradient.getValue().scalar() && muGradient.getValue().scalar() > -0.9);
-        Assert.assertTrue(1.8 > muIntercept.getValue().scalar() && muIntercept.getValue().scalar() > 1.2);
-
-        Assert.assertTrue(0.5 > sigmaGradient.getValue().scalar() && sigmaGradient.getValue().scalar() > 0.);
-        Assert.assertTrue(0.5 > sigmaIntercept.getValue().scalar() && sigmaIntercept.getValue().scalar() > 0.);
-
-        for (RegressionModel subModel : models) {
```

Comment:
Why are we no longer checking the values of the random variables of the children in the hierarchical model?

Reply:
added this back! Was just checking really core stuff first